### PR TITLE
🎉 Source Facebook Marketing: Add AdAccount and Images stream implementation

### LINKED ISSUE #6226: Facebook Marketing Source:  Add "Ad Image" stream.
## Tell us about the problem you're trying to solve
I am using the Facebook Marketing source connector, and would like to be able to see the performance of various ads next to the link of the associated image in the creative.  

In certain cases, the image url is contained in the Ad Creative object, which is one of the core streams synced with the Facebook Marketing connector.  

In many cases, however, that's not where the data is stored.

## Describe the solution you’d like

In the many cases where the image url of an ad creative is not available in the Ad Creative stream, it is because it is stored in the Ad Library and can be accessed with a different object through the Marketing API:  `Ad Image`

https://developers.facebook.com/docs/marketing-api/reference/ad-image

I would like to be able to sync the Ad Image stream along with the existing streams (Ad, Adset, Ad Creative, Campaign, and Ad Insights)

## Describe the alternative you’ve considered or used
I have not found a workaround for getting the data.

## Are you willing to submit a PR?

Not immediately

### LINKED ISSUE #9840: Facebook Marketing Source: Add "Ad Account" Stream
## Tell us about the problem you're trying to solve
Support Ad Account stream to reach feature parity for GA connector: https://developers.facebook.com/docs/marketing-api/reference/ad-account. Based off [MLP research](https://docs.google.com/document/d/1lQlxwLt22CFMAG3qTWYn4MpRSi2XCLQVqEkb0XeqBeA/edit#).

### LINKED ISSUE #9842: Facebook Marketing Source: Support "Replicate Incremental Deletes"
## Tell us about the problem you're trying to solve
Support replicating incremental deletes to reach parity for GA connector as per [research](https://docs.google.com/document/d/1lQlxwLt22CFMAG3qTWYn4MpRSi2XCLQVqEkb0XeqBeA/edit#).

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
#9842 - Facebook Marketing Source: Support "Replicate Incremental Deletes"
#9840 - Facebook Marketing Source: Add "Ad Account" stream
#6226 - Facebook Marketing Source: Add "Ad Image" stream

## How
Added `AdAccount` and `AdImage` streams implementation. 
Added to docs info about Incremental Deletes Sync.

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
  
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
+  "type": "object",
+  "additionalProperties": true,
+  "properties": {
+    "account_id": {
+      "type": "string"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json:6]: nit: ["null", "string"]

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    "id": {
+      "type": "string"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json:9]: nit: ["null", "string"]

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    "name": {
+      "type": "string"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json:12]: nit: ["null", "string"]

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    "creatives": {
+      "type": ["null", "array"],
+      "items": {
+        "type": "string"
```
Comment (bazarnov) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/schemas/images.json:17]: nit: ["null", "string"]

**Review by bazarnov — APPROVED** [review]
Please check the comments for schemas.

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py [hunk 5]
```diff
@@ -293,6 +297,82 @@ def _read_records(self, params: Mapping[str, Any]) -> Iterator:
...
+
+        potentially_new_records_in_the_past = self._include_deleted and not stream_state.get("include_deleted", False)
+        if potentially_new_records_in_the_past:
+            self.logger.info(f"Ignoring bookmark for {self.name} because of enabled `include_deleted` option")
+            filter_value = self._start_date
+
+        first_sync = not stream_state and str(self._start_date) <= record[self.cursor_field]
+
+        if first_sync or pendulum.parse(record[self.cursor_field]) > filter_value:
+            yield record
```
Comment (antixar) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py:361]: this generator returns one item or nothing anyways. Maybe can we convert to some checking function?
`yield from self.filter_by_state(stream_state=stream_state, record=record.__dict__["_json"])`
=>
```
if self.check_record_relevance(stream_state=stream_state, record=record.__dict__["_json"]):
    yield record
```

**Review by antixar — COMMENTED** [review]
several minor comments

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py (hunk 5 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py:361]: I think it is better to divide filtering logic from `read_records` method.

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py [hunk 6]
```diff
@@ -293,6 +297,82 @@ def _read_records(self, params: Mapping[str, Any]) -> Iterator:
...
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        """Main read method used by CDK"""
+        yield self._api.account.api_get(params=self.request_params(stream_state=stream_state)).__dict__["_json"]
+
```
[the remark this answers is not in the document]
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py:317]: Added comment in code

**Review by antixar — APPROVED** [review]
LGTM
Please publish your changes

**Pull request merged**